src/persist/src/indexed/mod.rs

```diff
@@ -66,9 +88,13 @@ pub struct Indexed<U: Buffer, L: Blob> {
     // Indexed or somewhere else.
     buf: U,
     blob: BlobCache<L>,
-    futures: HashMap<Id, BlobFuture>,
-    traces: HashMap<Id, BlobTrace>,
+    futures: BTreeMap<Id, BlobFuture>,
+    traces: BTreeMap<Id, BlobTrace>,
     listeners: HashMap<Id, Vec<ListenFn<Vec<u8>, Vec<u8>>>>,
+
+    pending_writes: Vec<(Id, Vec<((Vec<u8>, Vec<u8>), u64, isize)>)>,
```

Comment:
we should also have a comment somewhere, TBD where, putting together how all this works

Reply:
added those notes to those fields and added this doc comment under Notes above Indexed

```
/// - Writes, seals, and allow_compaction requests are not committed to durable
///   storage immediately because we want to amortize the cost of writing to
///   durable storage across many of those requests. Instead, those requests are
///   applied to the in-memory Indexed object, and their responses are stored in
///   `pending_responses`. Later pending writes and serialized metadata are
///   committed to durable storage in `drain_pending` which flushes all pending
///   writes to [BlobFuture], commits the current serialized metadata to durable
///   storage, and clears and responds to all pending responses.
```